src/client/api/event.c

```diff
@@ -328,6 +329,8 @@ daos_event_complete_locked(struct daos_eq_private *eqx,
 
 	if (eqx != NULL)
 		eq = daos_eqx2eq(eqx);
+	else
+		D_MUTEX_LOCK(&evx->evx_lock);
```

Comment:
seems D_MUTEX_INIT was call in the case of no eq, and no parent event?
here possibly use the evx_lock for the case of parent event valid?

Reply:
Yea the parent event usage is actually non existent and we should probably remove it since we just use tse for dependencies now. But yea I should just check if parent is null to take the lock

Reply:
ok now we init the mutex for child events where the parent eq is NULL, so the lock should be valid here